netbox/utilities/filters.py

```diff
@@ -111,6 +118,152 @@ def __init__(self, *args, **kwargs):
 # FilterSets
 #
 
+class BaseFilterSet(django_filters.FilterSet):
+    """
+    A base filterset which provides common functionaly to all NetBox filtersets
+    """
+    FILTER_DEFAULTS = deepcopy(django_filters.filterset.FILTER_FOR_DBFIELD_DEFAULTS)
```

Comment:
Why was this moved to inside the FilterSet?

Reply:
Today we are simply doing this:
```
FILTER_DEFAULTS = django_filters.filterset.FILTER_FOR_DBFIELD_DEFAULTS
```
At the root of the module, which is actually just a weak reference to the original dictionary which is then also referenced in this same way in `django_filters.FilterSet`. So while it does work today because of the global nature in which we use these defaults, it is a form of monkey patch. Overriding it in the base class is the "more correct" way to do it and would allow us to use alternate functionality in some other case should the need ever arise. It also seems to be the unofficial recommendation of django-filters when looking at other solutions where people are using base classes like this.